frontend/src/lib/DestinationForm.svelte

```diff
@@ -0,0 +1,120 @@
+            name: getFormValue(form, "name"),
+            settings: {}
+        };
+
+        const parameters = await connectionParamsResponse;
+
+        for (const { name } of parameters.form_fields) {
+            payload.settings[name] = getFormValue(form, name);
+        }
+
+        const resource = "/api/destinations";
+        const body = JSON.stringify(payload);
+        const options = {
+            method: "POST",
+            headers: {
+                "content-type": "application/json",
+                "content-length": body.length,
+            },
+            body
+        };
+
+        const response = await fetch(resource, options);
+
+        if (response.ok) {
+            const statusSeeOther = 303;
+            const location = "/dashboard/connections";
+            redirect(statusSeeOther, location);
+        } else {
+            const errorText = await response.text();
+
+            alertError.show(errorText);
+        }
+    }
+
+    let connectionParamsResponse = emptyParams();
+
+    export let alertError;
+    export let connectionType;
+    export let loaded = false;
+    export function load(name) {
+        if (name && name !== connectionType) {
+            loaded = false;
+            connectionType = name;
+        }
+
+        if (loaded === true) {
+            return;
+        }
+
+        connectionParamsResponse = getDestinationParameters(connectionType);
+        connectionParamsResponse.then(() => {
+            loaded = true;
+        });
+    }
+</script>
+
+{#await connectionParamsResponse}
+{:then parameters}
+    <form id="{connectionType}-form" on:submit|preventDefault="{saveNewDestination}">
+        <DestinationTextParam name="name" label="Connection Name" />
+        {#each parameters.form_fields as formField}
+            {#if formField.type === "text"}
+                <DestinationTextParam name="{formField.name}" label="{formField.label}" defaultValue="{formField.default}" />
+            {:else if formField.type === "password"}
+                <DestinationPasswordParam name="{formField.name}" label="{formField.label}" defaultValue="{formField.default}" />
+            {:else if formField.type === "bool"}
+                <DestinationBoolParam name="{formField.name}" label="{formField.label}" defaultValue="{formField.default}" />
+            {:else if formField.type === "number"}
+                <DestinationNumberParam name="{formField.name}" label="{formField.label}" defaultValue="{formField.default}" />
+            {:else if formField.type === "textarea"}
+                <DestinationTextareaParam name="{formField.name}" label="{formField.label}" defaultValue="{formField.default}" />
+            {/if}
+        {/each}
+        <br>
+        <sl-button type="submit" variant="primary" outline pill>
+            <sl-icon name="cloud-plus" slot="prefix"></sl-icon>
+            Save New Connection
+        </sl-button>
+        <sl-button type="reset" variant="warning" outline pill>
```

Comment:
I don't think we need a reset button here.

Reply:
A reset button is rarely necessary for typical users. It's more an accessibility utility.